src/o365/todo/commands/list/list-add.ts

```diff
@@ -0,0 +1,92 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  name: string;
+}
+
+class TodoListAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.LIST_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft To Do task list';
+  }
+
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const body: any = {
+      displayName: args.options.name
+    };
+
+
+    const requestOptions: any = {
+      url: `${this.resource}/beta/me/todo/lists`,
+      headers: {
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json'
+      },
+      body,
+      json: true
+    };
+
+    request
+      .post(requestOptions)
+      .then((res: any): void => {
+        cmd.log(res);
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-n, --name <name>',
+        description: `The name of the task list to add`
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.name) {
+        return 'Required option name is missing';
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: {}, log: (help: string) => void): void {
+    log(vorpal.find(this.name).helpInformation());
+    log(
+      `  
+  Examples:
+  
+    Add a list called "My task list"
+      ${this.name} --name "My task list"
+`);
+  }
```

Comment:
Fixed indentation, added `task` & replaced quotes in `"My task list"` with chalk, to be consistent with other commands such as `spo list add`.

```
    const chalk = vorpal.chalk;
    log(vorpal.find(this.name).helpInformation());
    log(
      ` Examples:

    Add a task list with the name ${chalk.grey('My task list')}
      ${this.name} --name "My task list"
      `);
```

Reply:
Damn... 😁 I took extra care to indent and chalk, I took another random in teams commands to do similar... but I am quite happy with such minor adjustments :) I’ll get to the point, some day, you guys will have nothing to fix :p

Reply:
I've just realised that we are missing a Remark section to state that the endpoint this command uses is in preview. I'll get this added before we merge.